salt/minion.py

```diff
@@ -1507,26 +1509,21 @@ def _handle_decoded_payload(self, data):
                     target=self._target, args=(instance, self.opts, data, self.connected)
                 )
         else:
-            process = threading.Thread(
-                target=self._target,
-                args=(instance, self.opts, data, self.connected),
-                name=data['jid']
-            )
+            self.thread_pool.submit(self._target, instance, self.opts, data, self.connected)
 
         if multiprocessing_enabled:
             with default_signals(signal.SIGINT, signal.SIGTERM):
                 # Reset current signals before starting the process in
                 # order not to inherit the current signal handlers
                 process.start()
-        else:
-            process.start()
 
         # TODO: remove the windows specific check?
-        if multiprocessing_enabled and not salt.utils.platform.is_windows():
-            # we only want to join() immediately if we are daemonizing a process
-            process.join()
-        elif salt.utils.platform.is_windows():
-            self.win_proc.append(process)
+        if multiprocessing_enabled:
```

Comment:
Line 1514 (above, originally 1516) already checks `if multiprocessing_enabled:`. Since there doesn't seem to be anything in between, you can just combine these 2 blocks.

Reply:
I agree, but this should have been done upstream too. Now I'm trying to avoid merge conflicts as much as possible, and that's why I tried to keep changes to a minimum